ansible/roles/flannel/tasks/client.yml

```diff
@@ -4,15 +4,33 @@
   args:
         name: flannel
         state: latest
-  when: not is_atomic and not is_coreos
+  when: not is_atomic and not is_coreos and ansible_distribution != "Ubuntu"
+
+- name: Force flannel_source_type to github when packages are not available
+  set_fact:
+    flannel_source_type: "github-release"
+  when: is_coreos or ansible_distribution == "Ubuntu"
 
 - name: Install from github
   include: github-release.yml
-  when: is_coreos
+  when: flannel_source_type == "github-release"
+
+- set_fact:
+    flannel_use_upstart: ansible_distribution == 'Ubuntu' and ansible_distribution_major_version|int < 15
 
 - name: Install Flannel config file
   template: src=flanneld.j2 dest=/etc/sysconfig/flanneld
```

Comment:
lets set a fact called 'flannel_config_dir'.  Default it to /etc/sysconfig. Overwrite it to /etc/flannel/ if you must. Then the follow on plays will need less duplication.